ci-automation/azure-sig.sh

```diff
@@ -0,0 +1,368 @@
+BLACKLISTED_TARGET_REGIONS=${BLACKLISTED_TARGET_REGIONS:-polandcentral australiacentral2 brazilsoutheast centraluseuap eastus2euap eastusstg francesouth germanynorth jioindiacentral norwaywest southafricawest switzerlandwest uaecentral brazilus southcentralusstg}
+DEFAULT_TARGET_REGIONS=$(az account list-locations -o json | jq -r '.[] | select( .metadata.regionType != "Logical" ) | .name' | sort | grep -v -E "(${BLACKLISTED_TARGET_REGIONS// /|})" | tr \\n ' ')
+TARGET_REGIONS=${TARGET_REGIONS:-${DEFAULT_TARGET_REGIONS}}
+
+# CAPI specific variables.
+KUBERNETES_SEMVER=${KUBERNETES_SEMVER:-v1.23.13}
+FLATCAR_CAPI_GALLERY_NAME=${FLATCAR_CAPI_GALLERY_NAME:-flatcar4capi}
+FLATCAR_CAPI_STAGING_GALLERY_NAME=${FLATCAR_CAPI_STAGING_GALLERY_NAME:-flatcar4capi_staging}
+FLATCAR_CAPI_IMAGE_NAME=${FLATCAR_CAPI_IMAGE_NAME:-${FLATCAR_IMAGE_NAME}-capi-${KUBERNETES_SEMVER}}
+FLATCAR_CAPI_IMAGE_OFFER=${FLATCAR_CAPI_IMAGE_OFFER:-${FLATCAR_CHANNEL}-capi}
+FLATCAR_CAPI_IMAGE_SKU=${FLATCAR_CAPI_IMAGE_SKU:-${FLATCAR_CAPI_IMAGE_NAME}}
+FLATCAR_CAPI_COMMUNITY_GALLERY_PUBLIC_NAME_PREFIX=${FLATCAR_CAPI_COMMUNITY_GALLERY_PUBLIC_NAME_PREFIX:-flatcar4capi}
+IMAGE_BUILDER_GIT_REMOTE="${IMAGE_BUILDER_GIT_REMOTE:-https://github.com/kubernetes-sigs/image-builder.git}"
+IMAGE_BUILDER_GIT_REPOSITORY_PATH="${IMAGE_BUILDER_GIT_REPOSITORY_PATH:-/tmp/image-builder}"
+IMAGE_BUILDER_GIT_VERSION="${IMAGE_BUILDER_GIT_VERSION:-main}"
+
+function publish-flatcar-capi-image() {
+  require-amd64-arch
+
+  # First, make sure staging image is available before publishing.
+  build-capi-staging-image
+
+  login
+
+  IMAGE_NAME="${FLATCAR_CAPI_IMAGE_NAME}"
+  IMAGE_VERSION="${FLATCAR_VERSION}"
+  GALLERY_NAME="${FLATCAR_CAPI_GALLERY_NAME}"
+  RESOURCE_GROUP_NAME="${PUBLISHING_SIG_RESOURCE_GROUP}"
+
+  # shellcheck disable=SC2310 # This might return 1.
+  if sig-image-version-exists; then
+    return
+  fi
+
+  ensure-resource-group
+  ensure-community-sig
+
+  IMAGE_OFFER="${FLATCAR_CAPI_IMAGE_OFFER}"
+  IMAGE_PUBLISHER="${IMAGE_PUBLISHER_NAME}"
+  ensure-image-definition
+
+  SOURCE_VERSION="/subscriptions/${AZURE_SUBSCRIPTION_ID}/resourceGroups/${STAGING_SIG_RESOURCE_GROUP}"
+  SOURCE_VERSION="${SOURCE_VERSION}/providers/Microsoft.Compute/galleries"
+  SOURCE_VERSION="${SOURCE_VERSION}/${FLATCAR_CAPI_STAGING_GALLERY_NAME}/images/${IMAGE_NAME}/versions/${IMAGE_VERSION}"
+
+  EXCLUDE_FROM_LATEST=true copy-sig-image-version
+}
+
+function build-capi-staging-image() {
+  require-amd64-arch
+
+  # First, make sure that base Flatcar image is available.
+  ensure-flatcar-staging-sig-image-version-from-vhd
+
+  login
+
+  IMAGE_NAME="${FLATCAR_CAPI_IMAGE_NAME}"
+  IMAGE_VERSION="${FLATCAR_VERSION}"
+  GALLERY_NAME="${FLATCAR_CAPI_STAGING_GALLERY_NAME}"
+  RESOURCE_GROUP_NAME="${STAGING_SIG_RESOURCE_GROUP}"
+
+  # shellcheck disable=SC2310 # This might return 1.
+  if sig-image-version-exists; then
+    return
+  fi
```

Comment:
This is not the wanted behavior when we have to rerun the release step. It should replace/republish.